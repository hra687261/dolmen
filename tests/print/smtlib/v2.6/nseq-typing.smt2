(set-logic ALL)

(declare-sort E 0)
(declare-const s (NSeq E))


(assert (= (nseq.first s) (nseq.last s)))
(assert (=
  (nseq.set s (nseq.first s) (nseq.get s (nseq.first s)))
  (nseq.const (nseq.first s) (nseq.last s) (nseq.get s (nseq.first s)))
))

(assert (= s (nseq.relocate s (nseq.first s))))
(assert (= s (nseq.concat s s)))
(assert (= s (nseq.slice s (nseq.first s) (nseq.last s))))
(assert (= s (nseq.update s s)))

(check-sat)