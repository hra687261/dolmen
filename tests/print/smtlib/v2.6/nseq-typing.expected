Warning The logic used to typecheck the problem (ALL) is not minimal. The minimal logic actually is: QF_UF
(set-logic QF_UF)
(declare-sort E 0)
(declare-const s (NSeq E))
(assert (= (nseq.first s) (nseq.last s)))
(assert (= (nseq.set s (nseq.first s) (nseq.get s (nseq.first s)))
         (nseq.const (nseq.first s) (nseq.last s)
          (nseq.get s (nseq.first s)))))
(assert (= s (nseq.relocate s (nseq.first s))))
(assert (= s (nseq.concat s s)))
(assert (= s (nseq.slice s (nseq.first s) (nseq.last s))))
(assert (= s (nseq.update s s)))
(check-sat)
(exit)
